crates/vm/levm/src/opcode_handlers/block.rs

```diff
@@ -160,21 +160,27 @@ impl VM {
     ) -> Result<OpcodeSuccess, VMError> {
         self.increase_consumed_gas(current_call_frame, gas_cost::BLOBHASH)?;
 
-        let Ok(index) = TryInto::<usize>::try_into(current_call_frame.stack.pop()?) else {
+        let index = current_call_frame.stack.pop()?;
+
+        let blob_hashes = &self.env.tx_blob_hashes;
+        if index > blob_hashes.len().into() {
             current_call_frame.stack.push(U256::zero())?;
             return Ok(OpcodeSuccess::Continue);
-        };
+        }
 
-        let blob_hashes = &self.env.tx_blob_hashes;
+        let index: usize = index
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
 
-        blob_hashes
-            .get(index)
-            .map(|el| {
-                current_call_frame
-                    .stack
-                    .push(U256::from_big_endian(el.as_bytes()))
-            })
-            .unwrap_or_else(|| current_call_frame.stack.push(U256::zero()))?;
+        if let Some(blob_hash) = blob_hashes.get(index) {
+            current_call_frame
+                .stack
+                .push(U256::from_big_endian(blob_hash.as_bytes()))?;
+        } else {
+            // This should never happen because whe check if the index
+            // fits inside the blobhash array above
+            return Err(VMError::Internal(InternalError::UndefinedState(1)));
+        }
```

Comment:
```suggestion
        let blob_hash = blob_hashes
    .get(index)
    .ok_or_else(|| VMError::Internal(InternalError::UndefinedState(1)))?;

current_call_frame
    .stack
    .push(U256::from_big_endian(blob_hash.as_bytes()))?;
```
IMO this looks better.
We don't want to use UndefinedState internal error inside opcodes, that is reserved for special occasions. Change it for another `InternalError`, you can add one for when we try to access a list with an index that is out of bounds.

Reply:
Noted Jere, I'll change it ASAP!

Reply:
Changed here: https://github.com/lambdaclass/ethrex/pull/1509/commits/b0613f9f9200a77ab0b79ffc541d8bf9267a5c31